### PR TITLE
Fix search loading errors

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #2934.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- load a search url directly e.g. /q/openstack/
- there should be no errors in your console

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7714/

File: jujugui/static/gui/src/app/components/search-results/search-results.js [hunk 1]
```diff
@@ -228,7 +228,11 @@ YUI.add('search-results', function(Y) {
       }
       this._changeActiveComponent('loading');
       this.setState({ waitingForSearch: true });
-
+      // If there is an exist search request then abort it before starting the
+      // new search.
+      if (this.searchXhr) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/search-results/search-results.js]: This is a good addition to have, but maybe some more investigation is necessary to find out why a pre-existing search is happening on the initial load, and that should be fixed.

Comment (huwshimi) [inline jujugui/static/gui/src/app/components/search-results/search-results.js]: This appears to get multiple state dispatches (like the other issues we had around this stuff) so it makes multiple searches.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7717/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7718/

**Comment by huwshimi** [comment]
Thanks @frankban and @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**